static/js/components/forms/RegisterDetailsForm.js

```diff
@@ -1,41 +1,296 @@
+      isValid,
+      values
+    }) => (
       <Form>
-        <label htmlFor="name">
-          Name
-          <Field type="text" name="name" />
-        </label>
-        <ErrorMessage name="name" component="div" />
-        <label htmlFor="password">
-          Password
-          <Field type="password" name="password" />
-        </label>
-        <ErrorMessage name="password" component="div" />
-        <button type="submit" disabled={isSubmitting || !isValid}>
-          Register
-        </button>
+        <div className="form-group">
+          <label htmlFor="legal_address.last_name" className="font-weight-bold">
+            Legal Name*
+          </label>
+          <Field type="text" name="name" className="form-control" />
+          <ErrorMessage name="name" component={FormError} />
+        </div>
+        <div className="form-group">
+          <label htmlFor="legal_address.first_name" className="row">
+            <div className="col-4 font-weight-bold">First Name*</div>
+            <div className="col-8">(Name that will appear on emails)</div>
+          </label>
+          <Field
+            type="text"
+            name="legal_address.first_name"
+            className="form-control"
+          />
+          <ErrorMessage name="legal_address.first_name" component={FormError} />
+        </div>
+        <div className="form-group">
+          <label htmlFor="legal_address.last_name" className="font-weight-bold">
+            Last Name*
+          </label>
+          <Field
+            type="text"
+            name="legal_address.last_name"
+            className="form-control"
+          />
+          <ErrorMessage name="legal_address.last_name" component={FormError} />
+        </div>
+        <div className="form-group">
+          <label htmlFor="password" className="font-weight-bold">
+            Password*
+          </label>
+          <Field type="password" name="password" className="form-control" />
+          <ErrorMessage name="password" component={FormError} />
+          <div className="label-secondary">
+            Password must contain:
+            <ul>
+              <li>at least 8 characters</li>
+              <li>at least 1 number and 1 letter</li>
+            </ul>
+          </div>
+        </div>
+        <div className="form-group">
+          {/* LegalAddress fields */}
+          <label htmlFor="legal_address.last_name" className="font-weight-bold">
+            Street Address*
+          </label>
+          <FieldArray
+            name="legal_address.street_address"
+            render={arrayHelpers => (
+              <div>
+                {values.legal_address.street_address.map((line, index) => (
+                  <div key={index}>
+                    <Field
+                      name={`legal_address.street_address[${index}]`}
+                      className="form-control row-inner"
+                    />
+                    {index === 0 ? (
+                      <ErrorMessage
```

Comment:
This error message needs some padding adjustments:
![Screenshot_2019-05-07 mitxpro(1)](https://user-images.githubusercontent.com/28598/57313575-e4b97500-70bd-11e9-86c3-56b5b755783b.png)